File: easy/13/csharp/Solution.cs
public class Solution {
    public int RomanToInt(string s)
    {
        Dictionary<char, int> map = new Dictionary<char, int>();
        int result = 0;
        
        map.Add('I', 1);
        map.Add('V', 5);
        map.Add('X', 10);
        map.Add('L', 50);
        map.Add('C', 100);
        map.Add('D', 500);
        map.Add('M', 1000);

        for (int i = 0; i < s.Length; i++)
        {
            int current = map[s[i]];
            
            if (i < s.Length - 1 && current < map[s[i + 1]]) {
                result -= current;
            } else {
                result += current;
            }
        }

        return result;
    }
    
    public static void Main(string[] args) {
        Solution sol = new Solution();
        Console.WriteLine(sol.RomanToInt("MCMXCIV"));  // Output: 58
    }
}

File: easy/13/cpp/Solution.cpp
#include <iostream>
#include <string>
#include <map>
using namespace std;

class Solution {
public:
    int romanToInt(string s) {
        map<char, int> m;
        int result = 0;

        m.insert({'I', 1});
        m.insert({'V', 5});
        m.insert({'X', 10});
        m.insert({'L', 50});
        m.insert({'C', 100});
        m.insert({'D', 500});
        m.insert({'M', 1000});

        for (int i = 0; i < s.length(); i++) {
            int current = m.at(s[i]);

            if (i < s.length() - 1 && current < m.at(s[i + 1])) {
                result -= current;
            }
            else {
                result += current;
            }
        }
        return result;
    }
};

int main() {
    Solution sol;
    std::cout << sol.romanToInt("MCMXCIV") << std::endl;
    return 0;
}

